tests/parametric/test_telemetry.py

```diff
@@ -89,6 +89,93 @@ def test_library_settings(self, library_env, test_agent, test_library):
+            pass
+        event = test_agent.wait_for_telemetry_event("app-started", wait_loops=400)
+        configuration = event["payload"]["configuration"]
+
+        configuration_by_name = {item["name"]: item for item in configuration}
+        for apm_telemetry_name, value in [
+            ("trace_rate_limit", "100"),
+            ("trace_header_tags", "header:tag"),
+            ("trace_enabled", ("true", True)),
+            ("trace_obfuscation_query_string_regexp", "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"),
+            ("trace_log_directory", "/some/temporary/directory"),
+            ("version", "123"),
+            ("trace_http_client_error_statuses", "400"),
+            ("trace_http_server_error_statuses", "500"),
+            ("trace_http_client_tag_query_string", ("true", True)),
+            (
+                "trace_client_ip_header",
+                "x-forwarded-for",
+            ),  # Unclear if correct key, see: https://docs.google.com/document/d/1kI-gTAKghfcwI7YzKhqRv2ExUstcHqADIWA4-TZ387o/edit?disco=AAABVcOUNfU
+            ("trace_service_mappings", "plugin:custom"),
+        ]:
+            if context.library == "cpp" and apm_telemetry_name in ("trace_header_tags"):
+                continue
+            apm_telemetry_name = _mapped_telemetry_name(context, apm_telemetry_name)
+            cfg_item = configuration_by_name.get(apm_telemetry_name)
+            assert cfg_item is not None, "Missing telemetry config item for '{}'".format(apm_telemetry_name)
+            if isinstance(value, tuple):
+                assert cfg_item.get("value") in value, "Unexpected value for '{}'".format(apm_telemetry_name)
+            else:
+                assert cfg_item.get("value") == value, "Unexpected value for '{}'".format(apm_telemetry_name)
+            assert cfg_item.get("origin") == "env_var", "Unexpected origin for '{}'".format(apm_telemetry_name)
+
+        # Golang and CPP do not support DD_TRACE_<INTEGRATION>_ENABLED, so don't test them for this config.
+        apm_telemetry_name = _mapped_telemetry_name(context, "trace_disabled_integrations")
+        cfg_item = configuration_by_name.get(apm_telemetry_name)
+        if (
+            context.library == "java"
+            or context.library == "dotnet"
+            or context.library == "node"
+            or context.library == "python"
+            or context.library == "ruby"
+        ):
+            assert cfg_item is not None, "Missing telemetry config item for '{}'".format(apm_telemetry_name)
+            assert cfg_item.get("value") is "grpc"
+        if context.library == "php":
+            assert cfg_item is not None, "Missing telemetry config item for '{}'".format(apm_telemetry_name)
+            assert cfg_item.get("value") is "phpredis"
+        # The trace_agent_url is a container address -- don't know the value, but we can assert its not empty (i.e, that it reports)
+        apm_telemetry_name = _mapped_telemetry_name(context, "trace_agent_url")
```

Comment:
This one is weird because there are multiple configurations that may combine to set the agent url (agent_host + agent_port OR agent_url). The telemetry behavior when nothing is set may be different for each language, so perhaps that is something we need to standardize on...

However, for the immediate term, may I propose that ONLY assert the configuration key `trace_agent_url` when we are actually explicitly setting environment variable `DD_TRACE_AGENT_URL`? WDYT?

Reply:
@link04 said there were issues with .NET when setting trace_agent_url in this test (that is, if it's not the actual agent address, the data is never received). Hence why I didn't set it/tried to assert over default behavior.

But in this test, I'm just asserting the telemetry exists, not asserting over its value. So this shouldn't be a concern?
>  there are multiple configurations that may combine to set the agent url (agent_host + agent_port OR agent_url)

Btw, setting DD_TRACE_AGENT_URL it to a random address still worked in Go, though (i.e, telemetry data was received).

Reply:
> And in this test, I'm just asserting the telemetry exists, not asserting over its value. So this shouldn't be a concern?

My worry is that not all languages may emit the trace_agent_url telemetry by default (maybe they report the host name and port?), but maybe we could leave this as-is and see

Reply:
Marking this as resolved since we decided all tracers should report telemetry for all "consistent config" items, regardless of whether they've been configured.